routes/mobileapplite1.js

```diff
@@ -0,0 +1,147 @@
+
+// shortcut
+var HTTPError = sUtil.HTTPError;
+
+
+/**
+ * The main router object
+ */
+var router = sUtil.router();
+
+/**
+ * The main application object reported when this module is require()d
+ */
+var app;
+
+
+/**
+ * A helper function that obtains the HTML from the MW API and
+ * loads it into a domino DOM document instance.
+ *
+ * @param {String} domain the domain to contact
+ * @param {Object} params an Object with all the query parameters for the MW API
+ * @return {Promise} a promise resolving as the HTML element object
+ */
+function apiGet(domain, params) {
+
+    // get the page from the MW API
+    return preq.get({
+        uri: 'http://' + domain + '/w/api.php',
+        query: params
+    }).then(function(callRes) {
+        // and then return the result
+        return BBPromise.resolve(callRes);
+    });
+
+}
+
+
+/**
+ * GET {domain}/v1/mobileapp/lite/{title}
+ * Gets the lite mobile app version of a given wiki page.
+ */
+router.get('/mobileapp/lite/:title', function(req, res) {
+
+    // get the page content from MW API mobileview
+    var apiParams = {
+        "action": "mobileview",
+        "format": "json",
+        "page": req.params.title,
+        "prop": "text|sections",
+        "sections": "all",
+        "sectionprop": "toclevel|line|anchor",
+        "noheadings": true,
+        "noimages": true
+    };
+
+    return apiGet(req.params.domain, apiParams)
+        // and then return it
+        .then(function(apiRes) {
+            // check if the query failed
+            if (apiRes.status > 299) {
+                // there was an error in the MW API, propagate that
+                throw new HTTPError({
+                    status: apiRes.status,
+                    type: 'api_error',
+                    title: 'MW API error',
+                    detail: apiRes.body
+                });
+            }
+
+            // transform all sections
+            var sections = apiRes.body.mobileview.sections;
+            for (var idx = 0; idx < sections.length; idx++) {
+                var section = sections[idx];
+                section.text = runDomTransforms(section.text);
+            }
+
+            res.status(200).type('json').end(JSON.stringify(apiRes.body.mobileview));
+            //res.status(200).type('json').end(util.inspect(apiRes.body.mobileview));
+            //res.status(200).type('html').end(apiRes.innerHTML);
+            //res.status(200).type('json').end(apiRes);
+        });
+
+});
+
+
+/**
+ * Nuke stuff from the DOM we don't want.
+ */
+function runDomTransforms(text) {
+
+    var doc = domino.createDocument(text);
+    rmSelectorAll(doc, 'div.noprint');
+    rmSelectorAll(doc, 'div.infobox');
+    rmSelectorAll(doc, 'div.hatnote');
+    rmSelectorAll(doc, 'div.metadata');
+    rmSelectorAll(doc, 'table'); // TODO: later we may want to transform some of the tables into a JSON structure
+    return doc.body.innerHTML;
+
+}
+
+
+function rmSelectorAll(doc, selector) {
```

Comment:
The function needs to be declared before `runDomTransforms()` (where it is used)